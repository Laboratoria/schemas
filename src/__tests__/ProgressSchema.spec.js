const mongoose = require('mongoose/browser');
const {
  ProgressSchema, UserSchema, CohortTopicSchema,
} = require('../..')(mongoose);

describe('ProgressSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, ProgressSchema);
    return doc.validate()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should pass at least with user and cohortTopic fields', () => {
    const user = new mongoose.Document({}, UserSchema);
    const cohortTopic = new mongoose.Document({}, CohortTopicSchema);
    const doc = new mongoose.Document({
      user: user._id,
      cohortTopic: cohortTopic._id,
    }, ProgressSchema);

    return doc.validate();
  });

  it('should pass with all fields', () => {
    const user = new mongoose.Document({}, UserSchema);
    const cohortTopic = new mongoose.Document({}, CohortTopicSchema);
    const doc = new mongoose.Document({
      user: user._id,
      cohortTopic: cohortTopic._id,
      completedDuration: 20,
      completedUnits: 5,
      percent: 20,
      completedAt: Date.now(),
    }, ProgressSchema);

    return doc.validate();
  });
});
