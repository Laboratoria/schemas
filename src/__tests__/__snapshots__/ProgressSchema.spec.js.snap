// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressSchema should fail validation when fields missing 1`] = `
Object {
  "cohortTopic": [ValidatorError: Path \`cohortTopic\` is required.],
  "user": [ValidatorError: Path \`user\` is required.],
}
`;
